### PR TITLE
fix preview dummy bug

### LINKED ISSUE #11556: Preview dummy shows clothing the species can't wear
## Description

E.g., shoes will be drawn on a species with no feet, even if it's fine once you get in-game.

Will affect https://github.com/space-wizards/space-station-14/pull/11497

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Closes #11556. Issue was just it literally never updated the dummy. I also removed a literally useless keeping track of whether the dummy should be updated.

**Screenshots**

Note: from my downstream because this doesn't effect any current wiz den species
image
Notice the lack of shoes now.

### CODE REVIEW

File: Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs [hunk 1]
```diff
@@ -839,16 +834,6 @@ private bool IsDirty
...
         }
 
-        private bool NeedsDummyRebuild
-        {
-            get => _needsDummyRebuild;
-            set
-            {
-                _needsDummyRebuild = value;
-                _needUpdatePreview = true;
-            }
```
Comment (mirrorcult) [inline Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs:849]: this wasnt useless if it was also setting `_needUpdatePreview`, right? all the instances where it was setting this before, it should be setting `_needUpdatePreview` to true as well

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs (hunk 1 above)
Comment (Elijahrane) [reply Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs:849]: done

**Review by wrexbe — CHANGES_REQUESTED** [review]
I merged this and the Diona PR together locally to see if the fix works, and Diona still has shoes.

**Comment by Elijahrane** [comment]
uhhh it was working before I added all those update functions

**Comment by Elijahrane** [comment]
Pulling the PR into my downstream to check against my species they still have no shoes, I think it might be an issue with the diona PR

**Comment by EmoGarbage404** [comment]
> Pulling the PR into my downstream to check against my species they still have no shoes, I think it might be an issue with the diona PR

I think that ui refactor may have bungled some stuff so i'll check that in a moment.

**Comment by EmoGarbage404** [comment]
ok, i think it's related to the fact that diona still have the actual feet body parts, just not the inventory slots.

**Comment by Elijahrane** [comment]
okay, I actually figured it out, it doesn't load the new species until you save the profile...

**Comment by Elijahrane** [comment]
image
saved profile

image
switch other profile to diona and don't save

**Comment by Elijahrane** [comment]
okay i got it i think

**Review by wrexbe — APPROVED** [review]

**Pull request merged**